### PR TITLE
Add `pkgdir`

### PR DESCRIPTION
**Pull request by omus**
Added to Base in: https://github.com/JuliaLang/julia/pull/33128

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 ## New functions, macros, and methods
 
+* `pkgdir(m)` return the root directory of the package that imported module `m` ([#33128]). (since Compat 2.3.0)
```
Comment (martinholters) [inline README.md]:
```suggestion
* `pkgdir(m)` returns the root directory of the package that imported module `m` ([#33128]).
```
(The next Compat version will be 3.0.0. And I propose to add the versions to the README when bumping the version in Project.toml, see https://github.com/JuliaLang/Compat.jl/issues/263#issuecomment-544988202. But if you prefer, you can also use 3.0.0 here now.)

Comment (fredrikekre) [reply README.md]: > The next Compat version will be 3.0.0

What changed?

Comment (martinholters) [reply README.md]: #665 which also removed old derprecations and introduced new ones.

Comment (martinholters) [reply README.md]: Well, by now it needs to be 3.1.0. I'll just commit the suggestion (not  setting the version number now) and we'll fix it when tagging the next version.

**Comment by martinholters** [comment]
Sorry this has fallen through the cracks. Let's move it forward now. And merry Christmas.

**Pull request merged**